docs/src/modules/java/pages/ask-akka/session.adoc

```diff
@@ -0,0 +1,267 @@
+
+== Overview
+
+This guide is concerned just with the session entity. We will get to more AI-specific tasks in the next steps.
+
+In this guide you will:
+
+* Create a new, empty Akka project
+* Create a data type for session events
+* Create a session entity
+* Create a conversation history view
+
+== Prerequisites
+
+include::ROOT:partial$cloud-dev-prerequisites.adoc[]
+
+== Create the empty project
+
+You already learned how to create an empty Akka project when you went through the guide to xref:java:author-your-first-service.adoc#_generate_and_build_the_project[author your first service]. Follow those steps again to ensure that you've got a new project that compiles.
+
+While you can use any settings you like, the code samples use:
+
+* Group ID: **akka.ask**
+* Artifact ID: **ask-akka**
+* Version: **1.0-SNAPSHOT**
+
+[NOTE]
+====
+This guide is written assuming you will follow it as a tutorial to walk through all of the components, building them on your own. If at any time you want to compare your solution with the official sample, check out the link:https://github.com/akka-samples/ask-akka-poc[Github repository].
+====
+
+
+== Create the session event
+
+Conversations, or sessions, with an LLM can be modeled nearly the same way as you might model the data from a regular chat application. You want to have the list of messages and, for each message, you want to know who supplied the message and when it was written. For LLM conversations, we also want to keep track of **token** usage because that may tie directly to how much a given session costs in real money.
+
+There are two types of participants in an LLM conversation:
+
+* The user
+* The AI (LLM)
+
+This means we can model a session event with `UserMessageAdded` and `AiMessageAdded` variants. While they have nearly identical structures now, it's worth keeping them as separate variants to allow them to expand separately in the future.
+
+Add a `SessionEvent.java` file to your `src/main/java/akka/ask/agent/domain/` directory (we're using the `agent` folder here because we cheated and know that we'll be creating other folders like `common` and `indexer` in later guides):
+
+[source, java, indent=0]
+----
+package akka.ask.agent.domain;
+
+import java.time.Instant;
+import akka.javasdk.annotations.TypeName;
+
+public sealed interface SessionEvent {
+
+  @TypeName("user-message-added")
+  public record UserMessageAdded(
+     String userId, 
+     String sessionId, 
+     String query, 
+     int tokensUsed, 
+     Instant timeStamp) implements SessionEvent {
+  }
+
+  @TypeName("ai-message-added") 
+  public record AiMessageAdded(
+     String userId, 
+     String sessionId, 
+     String response, // <1>
+     int tokensUsed, 
+     Instant timeStamp) implements SessionEvent {
+  }
+}
```

Comment:
We should always have code snippets in compiled code, otherwise it will become outdated and not work.
Shall we bring in the full sample here and mirror it to akka-samples as we do for the other samples?

Reply:
Yeah.. I think I'll copy that stuff in there for now. I think at some point in the not-do-distant future we should figure out a better strategy for maintaining samples without duplication and still have the docs be always up to date.